src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3983,5 +3983,36 @@ public void AssertNotDisposed()
             if (_disposeOne.Disposed)
                 ThrowObjectDisposed();
         }
+
+        public void Dump(string path, ref int totalNumberOfFiles, Action<string> currentFile)
+        {
+            if (Directory.Exists(path) == false)
+                Directory.CreateDirectory(path);
+
+            using (var tx = _environment.ReadTransaction())
+            {
+                var state = new Indexing.VoronState(tx);
+
+                var files = IndexPersistence.LuceneDirectory.ListAll(state);
+                totalNumberOfFiles = files.Length;
+                var buffer = new byte[64 * 1024];
+                foreach (var file in files)
+                {
+                    currentFile(file);
```

Comment:
Can we have here a message like: Exporting file: {file}.

Reply:
done